# 1. Unificación de csv

## Descripción:

La finalidad de este notebook es unir todos los datasets seleccionados en un único archivo.

## Librerías

In [1]:
import pandas as pd
import matplotlib.pyplot as plt

Veamos la versión de Python:

In [2]:
import sys
print(sys.version)

3.6.5 | packaged by conda-forge | (default, Apr  6 2018, 13:39:56) 
[GCC 4.8.2 20140120 (Red Hat 4.8.2-15)]


## Datasets

Cargamos los datasets en el notebook.

**EXCEL02**: *Dinámica poblacional. Datos de natalidad mortalidad, crecimiento vegetativo y tasas.*

In [3]:
excel02 = pd.read_csv('/home/jovyan//PRÁCTICAS DESPOBLAMIENTO/DATOS/NUEVOS DATOS/EXCEL_02.csv', sep=';')
excel02

CODMUN Provincia              Municipio  NAT2000  NAT2001  NAT2002  \
0       1001     Alava       Alegría-Dulantzi     29.0     25.0     27.0   
1       1002     Alava                Amurrio     72.0     66.0     54.0   
2       1003     Alava                Aramaio     15.0     18.0     15.0   
3       1004     Alava             Artziniega     12.0     13.0     12.0   
4       1006     Alava                Armiñón      1.0      4.0      0.0   
...      ...       ...                    ...      ...      ...      ...   
3851   50297  Zaragoza               Zaragoza   5464.0   5820.0   5952.0   
3852   50298  Zaragoza                  Zuera     54.0     30.0     47.0   
3853   50901  Zaragoza      Biel-Fuencalderas      1.0      1.0      0.0   
3854   50902  Zaragoza               Marracos      1.0      0.0      0.0   
3855   50903  Zaragoza  Villamayor de Gállego      NaN      NaN      NaN   

      NAT2003  NAT2004  NAT2005  NAT2006  ...    TCVEG2011   TCVEG2012  \
0        34.0     34.0     47.0       38  ...    11.477046   10.314595   
1        68.0     67.0     61.0       81  ...     1.298280    0.411997   
2        17.0     19.0     17.0       19  ...     3.731343   -6.226650   
3        20.0     12.0     17.0       17  ...     0.263104    0.144392   
4         3.0      2.0      2.0        5  ...    -0.526039    0.531915   
...       ...      ...      ...      ...  ...          ...         ...   
3851   6282.0   6357.0   6455.0     6708  ...  1088.695652  280.289331   
3852     59.0     46.0     51.0       64  ...   234.567901  153.846154   
3853      1.0      0.0      0.0        2  ...    -6.024096  -23.809524   
3854      0.0      0.0      0.0        1  ...     0.000000    0.000000   
3855      NaN      NaN      NaN       11  ...    -4.506066   -5.637773   

       TCVEG2013   TCVEG2014   TCVEG2015   TCVEG2016    TCVEG2017  \
0      11.105235   10.226050    6.571742    9.060023     4.592423   
1       2.015790    2.039429    2.423193    0.974659    -0.181373   
2       0.000000   14.511873   -2.751032   -4.213483     5.722461   
3       0.048445    0.036738   -0.024728    0.000000    -0.088035   
4       2.759382   -1.688239    0.565611   -1.146132    -2.302821   
...          ...         ...         ...         ...          ...   
3851  121.495327 -220.729367 -727.634195 -845.188285 -2010.706638   
3852   51.282051  289.156627   75.000000  381.578947     0.000000   
3853   -5.988024   -6.369427  -13.513514  -20.979021   -37.593985   
3854    0.000000  -20.618557  -21.276596  -32.967033     0.000000   
3855   -1.409940   -3.546099   -5.728607   -5.448602    -2.205072   

        TCVEG2018    TCVEG2019    TCVEG2020  
0        1.178550     7.888350     3.715170  
1       -0.925241    -1.978705    -1.248439  
2      -17.777778    -7.564297   -20.700637  
3       -0.076224    -0.038372    -0.025710  
4       -0.588235    -1.189768    -0.595593  
...           ...          ...          ...  
3851 -2723.175966 -2991.031390 -7393.103448  
3852   191.176471   148.648649  -146.067416  
3853     0.000000    -6.493506    -6.451613  
3854     0.000000   -34.883721   -23.809524  
3855     0.370920     1.102941    -1.457195  

[3856 rows x 150 columns]

Se van a eliminar las columnas de población por estar ya contenidas en el dataset excel05.

In [4]:
excel02 = excel02.loc[:, ~excel02.columns.str.startswith('POB')]
excel02

CODMUN Provincia              Municipio  NAT2000  NAT2001  NAT2002  \
0       1001     Alava       Alegría-Dulantzi     29.0     25.0     27.0   
1       1002     Alava                Amurrio     72.0     66.0     54.0   
2       1003     Alava                Aramaio     15.0     18.0     15.0   
3       1004     Alava             Artziniega     12.0     13.0     12.0   
4       1006     Alava                Armiñón      1.0      4.0      0.0   
...      ...       ...                    ...      ...      ...      ...   
3851   50297  Zaragoza               Zaragoza   5464.0   5820.0   5952.0   
3852   50298  Zaragoza                  Zuera     54.0     30.0     47.0   
3853   50901  Zaragoza      Biel-Fuencalderas      1.0      1.0      0.0   
3854   50902  Zaragoza               Marracos      1.0      0.0      0.0   
3855   50903  Zaragoza  Villamayor de Gállego      NaN      NaN      NaN   

      NAT2003  NAT2004  NAT2005  NAT2006  ...    TCVEG2011   TCVEG2012  \
0        34.0     34.0     47.0       38  ...    11.477046   10.314595   
1        68.0     67.0     61.0       81  ...     1.298280    0.411997   
2        17.0     19.0     17.0       19  ...     3.731343   -6.226650   
3        20.0     12.0     17.0       17  ...     0.263104    0.144392   
4         3.0      2.0      2.0        5  ...    -0.526039    0.531915   
...       ...      ...      ...      ...  ...          ...         ...   
3851   6282.0   6357.0   6455.0     6708  ...  1088.695652  280.289331   
3852     59.0     46.0     51.0       64  ...   234.567901  153.846154   
3853      1.0      0.0      0.0        2  ...    -6.024096  -23.809524   
3854      0.0      0.0      0.0        1  ...     0.000000    0.000000   
3855      NaN      NaN      NaN       11  ...    -4.506066   -5.637773   

       TCVEG2013   TCVEG2014   TCVEG2015   TCVEG2016    TCVEG2017  \
0      11.105235   10.226050    6.571742    9.060023     4.592423   
1       2.015790    2.039429    2.423193    0.974659    -0.181373   
2       0.000000   14.511873   -2.751032   -4.213483     5.722461   
3       0.048445    0.036738   -0.024728    0.000000    -0.088035   
4       2.759382   -1.688239    0.565611   -1.146132    -2.302821   
...          ...         ...         ...         ...          ...   
3851  121.495327 -220.729367 -727.634195 -845.188285 -2010.706638   
3852   51.282051  289.156627   75.000000  381.578947     0.000000   
3853   -5.988024   -6.369427  -13.513514  -20.979021   -37.593985   
3854    0.000000  -20.618557  -21.276596  -32.967033     0.000000   
3855   -1.409940   -3.546099   -5.728607   -5.448602    -2.205072   

        TCVEG2018    TCVEG2019    TCVEG2020  
0        1.178550     7.888350     3.715170  
1       -0.925241    -1.978705    -1.248439  
2      -17.777778    -7.564297   -20.700637  
3       -0.076224    -0.038372    -0.025710  
4       -0.588235    -1.189768    -0.595593  
...           ...          ...          ...  
3851 -2723.175966 -2991.031390 -7393.103448  
3852   191.176471   148.648649  -146.067416  
3853     0.000000    -6.493506    -6.451613  
3854     0.000000   -34.883721   -23.809524  
3855     0.370920     1.102941    -1.457195  

[3856 rows x 129 columns]

**EXCEL03**: *Migraciones. Estadísticas de Variaciones Residenciales y Saldos Migratorios a nivel municipal.*

In [5]:
excel03 = pd.read_csv('/home/jovyan/PRÁCTICAS DESPOBLAMIENTO/DATOS/NUEVOS DATOS/EXCEL_03.csv', sep=';')
excel03

CODMUN                 NOMBRE  ALTASTT2012  ALTASINT2012  ALTASEXT2012  \
0       1001       Alegría-Dulantzi          157           137            20   
1       1002                Amurrio          402           345            57   
2       1003                Aramaio           58            55             3   
3       1004             Artziniega           75            71             4   
4       1006                Armiñón           23            22             1   
...      ...                    ...          ...           ...           ...   
3851   50297               Zaragoza        17298         13085          4213   
3852   50298                  Zuera          349           277            72   
3853   50901                   Biel            4             4             0   
3854   50902               Marracos            3             3             0   
3855   50903  Villamayor de Gállego          105            99             6   

      BAJASTT2012  BAJASINT2012  BAJASEXT2012  SALDOTT2012  SALDOINT2012  ...  \
0             190           169            21          -33           -32  ...   
1             359           295            64           43            50  ...   
2              42            40             2           16            15  ...   
3             112            94            18          -37           -23  ...   
4              17            15             2            6             7  ...   
...           ...           ...           ...          ...           ...  ...   
3851        16156         12488          3668         1142           597  ...   
3852          284           195            89           65            82  ...   
3853            8             8             0           -4            -4  ...   
3854            7             7             0           -4            -4  ...   
3855          130           127             3          -25           -28  ...   

      SALDOEXT2019  ALTASTT2020  ALTASINT2020  ALTASEXT2020  BAJASTT2020  \
0               39          193           180            13          197   
1               37          373           312            61          308   
2                9           58            45            13           32   
3               21           90            84             6           77   
4                1           18            13             5           10   
...            ...          ...           ...           ...          ...   
3851          9186        16924         11012          5912        21753   
3852           -16          476           382            94          455   
3853             5           21            16             5            5   
3854             0           10            10             0            1   
3855            13          133           130             3          126   

      BAJASINT2020  BAJASEXT2020  SALDOTT2020  SALDOINT2020  SALDOEXT2020  
0              186            11           -4            -6             2  
1              255            53           65            57             8  
2               32             0           26            13            13  
3               67            10           13            17            -4  
4               10             0            8             3             5  
...            ...           ...          ...           ...           ...  
3851         14329          7424        -4829         -3317         -1512  
3852           295           160           21            87           -66  
3853             5             0           16            11             5  
3854             1             0            9             9             0  
3855           122             4            7             8            -1  

[3856 rows x 83 columns]

**EXCEL110**: *Estadísticas GIS territoriales. Municipios: estadísticas de altitud y pendiente en grados.*

In [6]:
excel110 = pd.read_csv('/home/jovyan/PRÁCTICAS DESPOBLAMIENTO/DATOS/NUEVOS DATOS/EXCEL 110_MUN_ALTITUDYPENDI.csv', sep=';')
excel110

CODMUNI  MINMDT  MAXMDT  RANGEMDT      MEANMDT      STDMDT  MINPENDI  \
0       33001     117    1394      1277   784.735210  233.924492  0.173360   
1       33002     271    2105      1834  1067.056635  382.986501  0.122584   
2       33003      99    2393      2294   817.504449  455.771315  0.346717   
3       33004      -6     131       137    41.899237   34.009653  0.000000   
4       33006     274    1113       839   506.502204  142.357363  0.122584   
...       ...     ...     ...       ...          ...         ...       ...   
3926    50293     671    1408       737   896.568072  140.151654  0.000000   
3927    50294     890     979        89   937.295656   17.358179  0.000000   
3928    50902     342     443       101   385.815187   19.015430  0.000000   
3929    50901     616    1443       827   870.803159  148.132527  0.122584   
3930    50903     201     461       260   306.537509   68.091967  0.000000   

       MAXPENDI  RANGEPENDI  MEANPENDI   STDPENDI  
0     46.501644   46.328284  20.298575   8.513073  
1     61.203892   61.081307  22.556711   8.204898  
2     67.186058   66.839341  25.832286  11.384265  
3     21.192459   21.192459   4.466160   3.501288  
4     43.220619   43.098035  15.395220   6.130982  
...         ...         ...        ...        ...  
3926  32.033096   32.033096   6.590386   4.319270  
3927  12.097475   12.097475   2.596466   2.142694  
3928  12.974638   12.974638   2.771706   2.017284  
3929  39.169685   39.047101  11.458718   6.380884  
3930  11.338626   11.338626   2.532540   1.777173  

[3931 rows x 11 columns]

**EXCEL111**: *Estadísticas GIS territoriales. Municipios: estadísticas de distancias a centros de servicios (núcleos >10.000 habitantes).*

In [7]:
excel111 = pd.read_csv('/home/jovyan/PRÁCTICAS DESPOBLAMIENTO/DATOS/NUEVOS DATOS/EXCEL 111_MUNICIPIOS_DISTANCIA A NUCMAS10MILHAB.csv', sep=';')
excel111

CODMUNI    MINDISN10M    MAXDISN10M  RANGEDISN10M   MEANDISN10M  \
0       33001  57553.715270  73094.160820  15540.445560  67040.405620   
1       33002   7258.173696  38852.515610  31594.341910  22574.525170   
2       33003  39241.563200  60094.631220  20853.068010  49222.642660   
3       33004     32.521914   4214.956742   4182.434829   2048.988149   
4       33006   7493.791483  16002.593660   8508.802179  11500.780800   
...       ...           ...           ...           ...           ...   
3926    50293  12814.442830  24312.267210  11497.824380  19346.719540   
3927    50294  32893.423320  37736.597950   4843.174623  35438.243400   
3928    50902  27809.314970  31199.314190   3389.999218  29703.772430   
3929    50901  24402.992640  36388.984140  11985.991500  31586.019460   
3930    50903   6157.843326  19642.635710  13484.792380  12558.205570   

       STDDISN10M  
0     3332.745043  
1     7310.916588  
2     5302.842339  
3      826.874570  
4     2088.300666  
...           ...  
3926  2769.176687  
3927  1153.859067  
3928   767.573192  
3929  2608.227995  
3930  3074.652190  

[3931 rows x 6 columns]

**EXCEL112**: *Estadísticas GIS territoriales. Municipios: estadísticas de distancias a infraestructuras.*

In [8]:
excel112 = pd.read_csv('/home/jovyan/PRÁCTICAS DESPOBLAMIENTO/DATOS/NUEVOS DATOS/EXCEL 112_MUNICIPIOS_DISTANCIAS_INFRAESTRUCTURAS.csv', sep=';')
excel112

CODMUNI    MINCARRETN    MAXCARRETN  RANGECARRETN   MEANCARRETN  \
0       33001  17472.278090  43871.030560  26398.752470  31120.853080   
1       33002   4114.285974  26602.411150  22488.125180  14322.659540   
2       33003     18.698440   8830.674861   8811.976421   3121.115962   
3       33004      8.882536   3030.249096   3021.366560   1206.657090   
4       33006   2057.640625   8594.165144   6536.524520   5528.735857   
...       ...           ...           ...           ...           ...   
3926    50293     22.287087   7750.586264   7728.299178   2906.384609   
3927    50294   1316.708524   5746.491560   4429.783036   3586.713091   
3928    50902  11094.715100  16069.586390   4974.871285  13736.181610   
3929    50901  16387.926510  31136.911890  14748.985380  23249.760590   
3930    50903   1070.642872  11415.170730  10344.527860   6309.529736   

       STDCARRETN     MINDISAUT     MAXDISAUT   RANGEDISAUT    MEANDISAUT  \
0     5415.753390  17539.891300  43926.642930  26386.751630  31623.687700   
1     5136.753584   2032.478440  35582.392350  33549.913910  17909.678160   
2     1753.653459  15257.027200  29760.796080  14503.768890  22003.181470   
3      639.815508     30.159684   5671.465399   5641.305715   1868.799672   
4     1482.212336   2937.367981  10618.578090   7681.210111   6812.411976   
...           ...           ...           ...           ...           ...   
3926  1658.470595  12280.893130  24995.670260  12714.777130  18954.536150   
3927  1079.835380    777.534817   5351.749461   4574.214644   3240.699440   
3928  1262.880413  11225.346580  16135.215700   4909.869116  13781.510770   
3929  3565.020569  17132.438290  31959.916450  14827.478160  24070.620200   
3930  2441.154293     17.385468  10875.739940  10858.354470   5576.955483   

        STDDISAUT    MINDISFERR    MAXDISFERR  RANGEDISFERR   MEANDISFERR  \
0     5996.385881  17866.633810  42724.783270  24858.149460  32835.065130   
1     7800.748080     13.116730  16521.288150  16508.171420   5313.377212   
2     3044.824209   7721.327493  26877.428850  19156.101350  16201.363920   
3     1152.691669     16.012535   3342.218867   3326.206333   1191.538806   
4     1978.469660   2238.611919   6705.702066   4467.090147   4517.535001   
...           ...           ...           ...           ...           ...   
3926  3085.342292  11830.585520  22110.095270  10279.509740  16677.655730   
3927  1049.837709   2618.815868   7192.330896   4573.515029   4861.438266   
3928  1229.917692  12846.490800  17492.221140   4645.730343  15103.415760   
3929  3570.692757   8890.815692  21516.737830  12625.922140  15677.993590   
3930  2512.526709   3065.242680  16180.075700  13114.833020   9150.936673   

       STDDISFERR  
0     5375.348561  
1     3528.866119  
2     4873.245249  
3      687.041000  
4     1028.025195  
...           ...  
3926  2380.118939  
3927  1110.592797  
3928  1046.244374  
3929  2972.601362  
3930  2792.152885  

[3931 rows x 16 columns]

**EXCEL04**: *Estructura. Evolución poblacional por quinquenios y sexos a nivel municipal.*

In [9]:
excel04 = pd.read_csv('/home/jovyan/PRÁCTICAS DESPOBLAMIENTO/DATOS/NUEVOS DATOS/EXCEL_04.csv', sep=';')
excel04

OBJECTID      NATCODE  CODMUN                   NOMBRE PROVINCIA  \
0            1  34033333001   33001                  Allande  Asturias   
1            2  34033333002   33002                    Aller  Asturias   
2            3  34033333003   33003                   Amieva  Asturias   
3            4  34033333004   33004                   Avilés  Asturias   
4            5  34033333006   33006                  Bimenes  Asturias   
...        ...          ...     ...                      ...       ...   
3926      3927  34025050293   50293  Villarroya de la Sierra  Zaragoza   
3927      3928  34025050294   50294     Villarroya del Campo  Zaragoza   
3928      3929  34025050902   50902                 Marracos  Zaragoza   
3929      3930  34025050901   50901                     Biel  Zaragoza   
3930      3931  34025050903   50903    Villamayor de Gállego  Zaragoza   

                        CCAA  0A4_21  0A4_10  0A4_03  5A9_21  ...   CARMAT21  \
0     Principado de Asturias    32.0    35.0    45.0    33.0  ...   8.866995   
1     Principado de Asturias   240.0   293.0   389.0   245.0  ...   7.352941   
2     Principado de Asturias     3.0    10.0    22.0    11.0  ...   1.351351   
3     Principado de Asturias  2187.0  3307.0  2800.0  2810.0  ...   9.117172   
4     Principado de Asturias    42.0    52.0    44.0    56.0  ...   9.600000   
...                      ...     ...     ...     ...     ...  ...        ...   
3926                  Aragón     6.0    11.0     8.0     7.0  ...   2.941176   
3927                  Aragón     2.0     1.0     0.0     4.0  ...   0.000000   
3928                  Aragón     2.0     1.0     3.0     1.0  ...   6.666667   
3929                  Aragón     2.0     0.0     2.0     2.0  ...   3.846154   
3930                  Aragón    98.0   112.0     0.0    82.0  ...  10.179641   

      POB2021  POB2010  POB2003  HOM2021  HOM2010  HOM2003  MUJ2021  MUJ2010  \
0        1620     2031   2296.0      800     1035   1191.0      820      996   
1       10346    12582  14366.0     5109     6190   7047.0     5237     6392   
2         616      805    883.0      353      426    471.0      263      379   
3       76874    84202  84182.0    36292    40352  40374.0    40582    43850   
4        1691     1894   2008.0      816      903    954.0      875      991   
...       ...      ...      ...      ...      ...      ...      ...      ...   
3926      445      565    632.0      236      297    327.0      209      268   
3927       89       75     78.0       48       42     39.0       41       33   
3928       92      100    119.0       49       51     61.0       43       49   
3929      171      186    213.0      102      108    120.0       69       78   
3930     2768     2888      0.0     1387     1437      0.0     1381     1451   

      MUJ2003  
0      1105.0  
1      7319.0  
2       412.0  
3     43808.0  
4      1054.0  
...       ...  
3926    305.0  
3927     39.0  
3928     58.0  
3929     93.0  
3930      0.0  

[3931 rows x 267 columns]

Se eliminan las variables relacionadas con la población, así como el código de natalidad (NATCODE) y las variables nombre, provincia y CCAA.

In [10]:
excel04 = excel04.loc[:, ~excel04.columns.str.startswith('POB')]
excel04 = excel04.loc[:, ~excel04.columns.str.startswith('HOM')]
excel04 = excel04.loc[:, ~excel04.columns.str.startswith('MUJ')]
excel04 = excel04.drop(['NATCODE'], axis=1)
excel04 = excel04.drop(['NOMBRE'], axis=1)
excel04 = excel04.drop(['PROVINCIA'], axis=1)
excel04 = excel04.drop(['CCAA'], axis=1)
excel04

OBJECTID  CODMUN  0A4_21  0A4_10  0A4_03  5A9_21  5A9_10  5A9_03  \
0            1   33001    32.0    35.0    45.0    33.0    36.0    56.0   
1            2   33002   240.0   293.0   389.0   245.0   322.0   445.0   
2            3   33003     3.0    10.0    22.0    11.0    16.0    16.0   
3            4   33004  2187.0  3307.0  2800.0  2810.0  3163.0  2854.0   
4            5   33006    42.0    52.0    44.0    56.0    41.0    48.0   
...        ...     ...     ...     ...     ...     ...     ...     ...   
3926      3927   50293     6.0    11.0     8.0     7.0    10.0    17.0   
3927      3928   50294     2.0     1.0     0.0     4.0     0.0     1.0   
3928      3929   50902     2.0     1.0     3.0     1.0     2.0     1.0   
3929      3930   50901     2.0     0.0     2.0     2.0     0.0     4.0   
3930      3931   50903    98.0   112.0     0.0    82.0   165.0     0.0   

      10A14_21  10A14_10  ...   HIJMUJ03    PRMEF10  MEFEC10   HIJMUJ10  \
0         34.0      49.0  ...   9.868421  64.457831      642   5.451713   
1        282.0     403.0  ...  12.712418  68.523154     4380   6.689498   
2         11.0      21.0  ...  14.864865  57.783641      219   4.566210   
3       3184.0    3017.0  ...  14.925373  74.155074    32517  10.170065   
4         42.0      53.0  ...  12.500000  62.462159      619   8.400646   
...        ...       ...  ...        ...        ...      ...        ...   
3926      10.0      21.0  ...   9.302326  53.731343      144   7.638889   
3927       1.0       1.0  ...   0.000000  39.393939       13   7.692308   
3928       1.0       3.0  ...  13.043478  75.510204       37   2.702703   
3929       1.0       3.0  ...   5.882353  56.410256       44   0.000000   
3930     121.0     159.0  ...   0.000000  75.603032     1097  10.209663   

        PRMEF21  MEFEC21   HIJMUJ21   CARMAT03  CARMAT10   CARMAT21  
0     24.756098      203  15.763547   4.166667  2.647975   8.866995  
1     29.864426     1564  15.345269   5.980392  3.059361   7.352941  
2     28.136882       74   4.054054   8.783784  2.739726   1.351351  
3     29.000542    11769  18.582717   7.164179  4.926654   9.117172  
4     28.571429      250  16.800000   6.818182  4.361874   9.600000  
...         ...      ...        ...        ...       ...        ...  
3926  32.535885       68   8.823529   2.325581  2.777778   2.941176  
3927  36.585366       15  13.333333   0.000000  7.692308   0.000000  
3928  34.883721       15  13.333333  13.043478  0.000000   6.666667  
3929  37.681159       26   7.692308   2.941176  0.000000   3.846154  
3930  36.278059      501  19.560878   0.000000  5.104831  10.179641  

[3931 rows x 253 columns]

**EXCEL05**: *Distribución de la población.  Cálculo de densidades a partir de los datos de evolución.*

In [11]:
excel05 = pd.read_csv('/home/jovyan/PRÁCTICAS DESPOBLAMIENTO/DATOS/NUEVOS DATOS/EXCEL_05.csv', sep=';')
excel05

OBJECTID  CODMUN  CCAA                   NOMBRE  CODPROV PROVINCIA  \
0            1   33001     3                  Allande       33  Asturias   
1            2   33002     3                    Aller       33  Asturias   
2            3   33003     3                   Amieva       33  Asturias   
3            4   33004     3                   Avilés       33  Asturias   
4            5   33006     3                  Bimenes       33  Asturias   
...        ...     ...   ...                      ...      ...       ...   
3926      3927   50293     2  Villarroya de la Sierra       50  Zaragoza   
3927      3928   50294     2     Villarroya del Campo       50  Zaragoza   
3928      3929   50902     2                 Marracos       50  Zaragoza   
3929      3930   50901     2                     Biel       50  Zaragoza   
3930      3931   50903     2    Villamayor de Gállego       50  Zaragoza   

                  NombreCCAA  Shape_Length    Shape_Area NombreProv  ...  \
0     Principado de Asturias  118035.51830  3.425453e+08   Asturias  ...   
1     Principado de Asturias   94295.88668  3.764015e+08   Asturias  ...   
2     Principado de Asturias   56655.78704  1.150417e+08   Asturias  ...   
3     Principado de Asturias   26157.82259  2.681264e+07   Asturias  ...   
4     Principado de Asturias   25335.32226  3.254734e+07   Asturias  ...   
...                      ...           ...           ...        ...  ...   
3926                  Aragón   38955.16492  9.161393e+07   Zaragoza  ...   
3927                  Aragón   15697.05947  1.695870e+07   Zaragoza  ...   
3928                  Aragón   20021.11297  1.697137e+07   Zaragoza  ...   
3929                  Aragón   51387.76143  1.305824e+08   Zaragoza  ...   
3930                  Aragón   42310.73582  8.923986e+07   Zaragoza  ...   

          DEN2012      DEN2013      DEN2014      DEN2015      DEN2016  \
0        5.660566     5.520438     5.424101     5.330682     5.155523   
1       32.242170    31.631121    31.264491    30.698606    29.983943   
2        6.980080     6.710613     6.588918     6.319450     6.189062   
3     3099.545896  3079.443435  3045.541511  3016.488047  2987.919428   
4       57.762023    55.672758    54.597401    54.320881    53.614218   
...           ...          ...          ...          ...          ...   
3926     6.036200     5.839723     5.686908     5.490431     5.217547   
3927     4.599408     4.599408     4.894242     4.717342     4.481475   
3928     6.245813     5.951199     5.715508     5.538740     5.361972   
3929     1.286544     1.278886     1.202306     1.133384     1.095094   
3930    31.801935    31.790729    31.600231    31.297676    30.849446   

          DEN2017      DEN2018      DEN2019      DEN2020      DEN2021  
0        5.085459     4.954090     4.811043     4.714705     4.729302  
1       29.295848    28.714023    28.195959    27.664611    27.486610  
2        6.076060     5.867440     5.745745     5.458892     5.354582  
3     2965.541921  2935.742539  2915.863853  2901.281177  2867.080886  
4       53.368422    52.231616    51.647851    51.586402    51.955096  
...           ...          ...          ...          ...          ...  
3926     5.097478     5.086563     4.868255     4.748186     4.857340  
3927     4.186641     4.009740     4.363541     5.248043     5.248043  
3928     5.244126     5.067358     5.067358     4.949512     5.420895  
3929     1.018514     1.072120     1.179332     1.186990     1.309518  
3930    30.490862    30.210718    30.479656    30.759800    31.017532  

[3931 rows x 85 columns]

**EXCEL12**: *Estadísticas GIS territoriales. Municipios: estadísticas de distancias a infraestructuras.*

In [12]:
excel12 = pd.read_csv('/home/jovyan/PRÁCTICAS DESPOBLAMIENTO/DATOS/NUEVOS DATOS/EXCEL_12.csv', sep=';')
excel12

codMuni  CCAA                                             Nombre  \
0        1001    16                                   Alegría-Dulantzi   
1        1002    16                                            Amurrio   
2        1003    16                                            Aramaio   
3        1004    16                                         Artziniega   
4        1006    16                                            Armiñón   
...       ...   ...                                                ...   
3926    53077     2                                          El Franco   
3927    53078    15                                    Bardenas Reales   
3928    53080    15                                   Sierra de Aralar   
3929    53081    15                                    Sierra de Lokiz   
3930    53083    15  Facería de Cogullo Alto, Cogullo Bajo, Sarmind...   

      CodProv NombreProvincia                  NombreCCAA  NHOSPI  NFARMA  \
0           1     Araba/Álava                  País Vasco       0       1   
1           1     Araba/Álava                  País Vasco       0       4   
2           1     Araba/Álava                  País Vasco       0       1   
3           1     Araba/Álava                  País Vasco       0       2   
4           1     Araba/Álava                  País Vasco       0       0   
...       ...             ...                         ...     ...     ...   
3926       53               0                      Aragón       0       0   
3927       53               0  Comunidad Foral de Navarra       0       0   
3928       53               0  Comunidad Foral de Navarra       0       0   
3929       53               0  Comunidad Foral de Navarra       0       0   
3930       53               0  Comunidad Foral de Navarra       0       0   

      NCENTEDUC  NATURGHOSP  NATPRIMARIA    DISTHOSPIT     DISTFARMA  \
0             2           0            1  13243.117270   8105.104541   
1             6           1            1  24308.144100  11466.318650   
2             0           0            1   6041.577103   9441.041825   
3             0           0            1  21149.641300   1301.885496   
4             0           0            0  10805.474600   2394.581647   
...         ...         ...          ...           ...           ...   
3926          0           0            0  63486.774090  16831.909780   
3927          0           0            0  18278.073010   5778.008141   
3928          0           0            0  15988.955630  17260.598190   
3929          0           0            0  12670.960150  21405.343920   
3930          0           0            0  10991.623830   4187.813121   

         DISTCEDUC     DISTCURGH   DISTATPRIM  
0      2609.754306   9790.487884  2661.540327  
1      3727.201496   5442.524424  3776.221905  
2      4271.741105   8713.460459  2902.566910  
3      8617.749843   8391.313642  2101.739841  
4      3950.406180   6468.294483  3511.013481  
...            ...           ...          ...  
3926  14973.750810  11866.083980  3681.207336  
3927  12424.164110  10784.092780  7739.682155  
3928   5149.795092   7167.692374  5206.291430  
3929  12987.121210   5969.911237  3600.443463  
3930  10388.515710   6749.366156  2129.977447  

[3931 rows x 16 columns]

Se eliminan variables no numericas, que, además, ya están contenidas en otros datasets.

In [13]:
excel12 = excel12.drop(['Nombre'], axis=1)
excel12 = excel12.drop(['CodProv'], axis=1)
excel12 = excel12.drop(['NombreProvincia'], axis=1)
excel12 = excel12.drop(['CCAA'], axis=1)
excel12

codMuni                  NombreCCAA  NHOSPI  NFARMA  NCENTEDUC  \
0        1001                  País Vasco       0       1          2   
1        1002                  País Vasco       0       4          6   
2        1003                  País Vasco       0       1          0   
3        1004                  País Vasco       0       2          0   
4        1006                  País Vasco       0       0          0   
...       ...                         ...     ...     ...        ...   
3926    53077                      Aragón       0       0          0   
3927    53078  Comunidad Foral de Navarra       0       0          0   
3928    53080  Comunidad Foral de Navarra       0       0          0   
3929    53081  Comunidad Foral de Navarra       0       0          0   
3930    53083  Comunidad Foral de Navarra       0       0          0   

      NATURGHOSP  NATPRIMARIA    DISTHOSPIT     DISTFARMA     DISTCEDUC  \
0              0            1  13243.117270   8105.104541   2609.754306   
1              1            1  24308.144100  11466.318650   3727.201496   
2              0            1   6041.577103   9441.041825   4271.741105   
3              0            1  21149.641300   1301.885496   8617.749843   
4              0            0  10805.474600   2394.581647   3950.406180   
...          ...          ...           ...           ...           ...   
3926           0            0  63486.774090  16831.909780  14973.750810   
3927           0            0  18278.073010   5778.008141  12424.164110   
3928           0            0  15988.955630  17260.598190   5149.795092   
3929           0            0  12670.960150  21405.343920  12987.121210   
3930           0            0  10991.623830   4187.813121  10388.515710   

         DISTCURGH   DISTATPRIM  
0      9790.487884  2661.540327  
1      5442.524424  3776.221905  
2      8713.460459  2902.566910  
3      8391.313642  2101.739841  
4      6468.294483  3511.013481  
...            ...          ...  
3926  11866.083980  3681.207336  
3927  10784.092780  7739.682155  
3928   7167.692374  5206.291430  
3929   5969.911237  3600.443463  
3930   6749.366156  2129.977447  

[3931 rows x 12 columns]

**Excel 11:** Viviendas turísticas, plazas y plazas por vivienda turística por municipios en 2021.

In [14]:
excel11 = pd.read_csv('/home/jovyan/PRÁCTICAS DESPOBLAMIENTO/DATOS/NUEVOS DATOS/EXCEL_11.csv', sep=';')
excel11

CODPROV  CODMUN                 NOMBRE  VIVTUR21  PLA21  PLAVT21
0           1    1001       Alegría-Dulantzi         3     14     4.67
1           1    1002                Amurrio         7     42     6.00
2           1    1003                Aramaio         0      0     0.00
3           1    1004             Artziniega         5     32     6.40
4           1    1006                Armiñón         2     14     7.00
...       ...     ...                    ...       ...    ...      ...
3851       50   50297               Zaragoza       659   2614     3.97
3852       50   50298                  Zuera         1      2     2.00
3853       50   50901                   Biel         1      4     4.00
3854       50   50902               Marracos         0      0     0.00
3855       50   50903  Villamayor de Gállego         0      0     0.00

[3856 rows x 6 columns]

In [15]:
excel11 = excel11.drop(['CODPROV'], axis=1)
excel11 = excel11.drop(['NOMBRE'], axis=1)
excel11 = excel11.drop(['PLAVT21'], axis=1)
excel11

CODMUN  VIVTUR21  PLA21
0       1001         3     14
1       1002         7     42
2       1003         0      0
3       1004         5     32
4       1006         2     14
...      ...       ...    ...
3851   50297       659   2614
3852   50298         1      2
3853   50901         1      4
3854   50902         0      0
3855   50903         0      0

[3856 rows x 3 columns]

**Excel 07:** Conexión digital. Zonas blancas y grises NGA. Variables a nivel de municipio.

In [16]:
excel07 = pd.read_csv('/home/jovyan/PRÁCTICAS DESPOBLAMIENTO/DATOS/NUEVOS DATOS/EXCEL_07.csv', sep=';')
excel07

CODNUCL  codMuni  CCAA                   NOMBRE  CodProv  PROVINCIA  \
0     34070505181     5181     7              Pascualcobo        5      Ávila   
1     34070505266     5266     7   Zapardiel de la Cañada        5      Ávila   
2     34073737176    37176     7                 Madroñal       37  Salamanca   
3     34033333001    33001     3                  Allande       33   Asturias   
4     34033333002    33002     3                    Aller       33   Asturias   
...           ...      ...   ...                      ...      ...        ...   
3926  34025050295    50295     2               Vistabella       50   Zaragoza   
3927  34025050298    50298     2                    Zuera       50   Zaragoza   
3928  34025050293    50293     2  Villarroya de la Sierra       50   Zaragoza   
3929  34025050294    50294     2     Villarroya del Campo       50   Zaragoza   
3930  34025050902    50902     2                 Marracos       50   Zaragoza   

                  NombreCCAA     AREAMUN AFECBLAN AFECGRIS  ...     PRPZB21  \
0            Castilla y León   16.137923       NO       SI  ...    0.000000   
1            Castilla y León   40.348615       SI       NO  ...  100.000000   
2            Castilla y León    1.593758       SI       SI  ...    0.000000   
3     Principado de Asturias  342.545260       SI       NO  ...    0.000000   
4     Principado de Asturias  376.401454       SI       NO  ...    0.437096   
...                      ...         ...      ...      ...  ...         ...   
3926                  Aragón   21.846607       SI       SI  ...    0.000000   
3927                  Aragón  332.207477       SI       SI  ...    0.000000   
3928                  Aragón   91.613932       SI       SI  ...    0.000000   
3929                  Aragón   16.958704       NO       NO  ...    0.000000   
3930                  Aragón   16.971368       SI       SI  ...    0.000000   

       Shape_Length    Shape_Area  POB21_ZG  CUEN_ZG    POB21  CBMAS30  \
0      24207.556470  1.613792e+07        41        1     41.0     0.00   
1      28466.935730  4.034861e+07         0        2     85.0     0.54   
2       6667.707807  1.593758e+06       136        1    136.0     1.00   
3     118035.518300  3.425453e+08         0        0   2032.0     0.00   
4      94295.886680  3.764015e+08         0        0  10524.0     1.00   
...             ...           ...       ...      ...      ...      ...   
3926   19939.156460  2.184661e+07         0        0     51.0     0.95   
3927  105328.960100  3.322075e+08         0        0   8626.0     0.95   
3928   38955.164920  9.161393e+07         0        0    445.0     0.86   
3929   15697.059470  1.695870e+07         0        0     89.0     0.95   
3930   20021.112970  1.697137e+07         0        0     96.0     0.95   

      CBMAS100  COB3G  COB4G  
0         0.00   1.00   0.90  
1         0.00   0.57   0.94  
2         0.00   1.00   1.00  
3         0.00   0.96   0.94  
4         0.00   1.00   1.00  
...        ...    ...    ...  
3926      0.00   0.98   1.00  
3927      0.02   1.00   1.00  
3928      0.86   1.00   1.00  
3929      0.00   1.00   1.00  
3930      0.00   1.00   1.00  

[3931 rows x 29 columns]

Eliminación de variables:

In [17]:
excel07 = excel07.drop(['CODNUCL'], axis=1)
excel07 = excel07.drop(['codMuni'], axis=1)
excel07 = excel07.drop(['CodProv'], axis=1)
excel07 = excel07.drop(['CCAA'], axis=1)
excel07 = excel07.drop(['NOMBRE'], axis=1)
excel07 = excel07.drop(['PROVINCIA'], axis=1)
excel07 = excel07.drop(['NombreCCAA'], axis=1)
excel07 = excel07.drop(['AREAMUN'], axis=1)
excel07 = excel07.drop(['AFECBLAN'], axis=1)
excel07 = excel07.drop(['AFECGRIS'], axis=1)
excel07 = excel07.drop(['Shape_Length'], axis=1)
excel07 = excel07.drop(['Shape_Area_1'], axis=1)
excel07 = excel07.drop(['Shape_Length_1'], axis=1)
excel07 = excel07.drop(['Shape_Area'], axis=1)
excel07 = excel07.drop(['POB21_ZB'], axis=1)
excel07 = excel07.drop(['POB21_ZG'], axis=1)
excel07 = excel07.drop(['POB21'], axis=1)
excel07 = excel07.drop(['CUEN_ZB'], axis=1)
excel07 = excel07.drop(['CUEN_ZG'], axis=1)
excel07 = excel07.drop(['PRMUNZB'], axis=1)
excel07 = excel07.drop(['PRMUNZG'], axis=1)
excel07 = excel07.drop(['PRPZB21'], axis=1)
excel07

CODMUN   KM2BLAN  KM2GRIS  CBMAS30  CBMAS100  COB3G  COB4G
0       5181  0.000000        0     0.00      0.00   1.00   0.90
1       5266  0.003671        0     0.54      0.00   0.57   0.94
2      37176  0.000429        0     1.00      0.00   1.00   1.00
3      33001  0.001683        0     0.00      0.00   0.96   0.94
4      33002  0.026321        0     1.00      0.00   1.00   1.00
...      ...       ...      ...      ...       ...    ...    ...
3926   50295  0.002940        0     0.95      0.00   0.98   1.00
3927   50298  0.415539        0     0.95      0.02   1.00   1.00
3928   50293  0.000139        0     0.86      0.86   1.00   1.00
3929   50294  0.000000        0     0.95      0.00   1.00   1.00
3930   50902  0.000650        0     0.95      0.00   1.00   1.00

[3931 rows x 7 columns]

## Unificación de CSV:

Como los csv excel110, excel111 y excel112 son similares, se va a comenzar uniendo estos tres archivos en uno. Primero, comprobemos que la columna del código del municipio son iguales para excel110 y excel111.

In [18]:
print(excel110['CODMUNI'] == excel111['CODMUNI'])

0       True
1       True
2       True
3       True
4       True
        ... 
3926    True
3927    True
3928    True
3929    True
3930    True
Name: CODMUNI, Length: 3931, dtype: bool


Se comprueba que todos los valores de esa columna estén contenidos en los dos csv.

In [19]:
print(excel110['CODMUNI'][(excel110['CODMUNI'] == excel111['CODMUNI']) == False])

Series([], Name: CODMUNI, dtype: int64)


Todos los valores están contenidos en la columna porque no sale ningún valor en la lista.

In [20]:
print(excel110['CODMUNI'][(excel110['CODMUNI'] == excel111['CODMUNI']) == True])

0       33001
1       33002
2       33003
3       33004
4       33006
        ...  
3926    50293
3927    50294
3928    50902
3929    50901
3930    50903
Name: CODMUNI, Length: 3931, dtype: int64


Repetimos la misma operación para el archivo excel110 y excel 112.

In [21]:
print(excel110['CODMUNI'] == excel112['CODMUNI'])

0       True
1       True
2       True
3       True
4       True
        ... 
3926    True
3927    True
3928    True
3929    True
3930    True
Name: CODMUNI, Length: 3931, dtype: bool


In [22]:
print(excel110['CODMUNI'][(excel110['CODMUNI'] == excel112['CODMUNI']) == False])

Series([], Name: CODMUNI, dtype: int64)


Encontramos los mismos resultados que en el caso anterior, por lo que se puede concluir que las columnas de los tres archivos son iguales.

In [23]:
print(excel110['CODMUNI'][(excel110['CODMUNI'] == excel112['CODMUNI']) == True])

0       33001
1       33002
2       33003
3       33004
4       33006
        ...  
3926    50293
3927    50294
3928    50902
3929    50901
3930    50903
Name: CODMUNI, Length: 3931, dtype: int64


Unimos los tres archivos en una mismo csv por la columna CODMUNI.

In [24]:
excel110_111 = pd.merge(excel110, excel111, on='CODMUNI')
excel110_112 = pd.merge(excel110_111, excel112, on='CODMUNI')

In [25]:
excel110_112

CODMUNI  MINMDT  MAXMDT  RANGEMDT      MEANMDT      STDMDT  MINPENDI  \
0       33001     117    1394      1277   784.735210  233.924492  0.173360   
1       33002     271    2105      1834  1067.056635  382.986501  0.122584   
2       33003      99    2393      2294   817.504449  455.771315  0.346717   
3       33004      -6     131       137    41.899237   34.009653  0.000000   
4       33006     274    1113       839   506.502204  142.357363  0.122584   
...       ...     ...     ...       ...          ...         ...       ...   
3926    50293     671    1408       737   896.568072  140.151654  0.000000   
3927    50294     890     979        89   937.295656   17.358179  0.000000   
3928    50902     342     443       101   385.815187   19.015430  0.000000   
3929    50901     616    1443       827   870.803159  148.132527  0.122584   
3930    50903     201     461       260   306.537509   68.091967  0.000000   

       MAXPENDI  RANGEPENDI  MEANPENDI  ...     MINDISAUT     MAXDISAUT  \
0     46.501644   46.328284  20.298575  ...  17539.891300  43926.642930   
1     61.203892   61.081307  22.556711  ...   2032.478440  35582.392350   
2     67.186058   66.839341  25.832286  ...  15257.027200  29760.796080   
3     21.192459   21.192459   4.466160  ...     30.159684   5671.465399   
4     43.220619   43.098035  15.395220  ...   2937.367981  10618.578090   
...         ...         ...        ...  ...           ...           ...   
3926  32.033096   32.033096   6.590386  ...  12280.893130  24995.670260   
3927  12.097475   12.097475   2.596466  ...    777.534817   5351.749461   
3928  12.974638   12.974638   2.771706  ...  11225.346580  16135.215700   
3929  39.169685   39.047101  11.458718  ...  17132.438290  31959.916450   
3930  11.338626   11.338626   2.532540  ...     17.385468  10875.739940   

       RANGEDISAUT    MEANDISAUT    STDDISAUT    MINDISFERR    MAXDISFERR  \
0     26386.751630  31623.687700  5996.385881  17866.633810  42724.783270   
1     33549.913910  17909.678160  7800.748080     13.116730  16521.288150   
2     14503.768890  22003.181470  3044.824209   7721.327493  26877.428850   
3      5641.305715   1868.799672  1152.691669     16.012535   3342.218867   
4      7681.210111   6812.411976  1978.469660   2238.611919   6705.702066   
...            ...           ...          ...           ...           ...   
3926  12714.777130  18954.536150  3085.342292  11830.585520  22110.095270   
3927   4574.214644   3240.699440  1049.837709   2618.815868   7192.330896   
3928   4909.869116  13781.510770  1229.917692  12846.490800  17492.221140   
3929  14827.478160  24070.620200  3570.692757   8890.815692  21516.737830   
3930  10858.354470   5576.955483  2512.526709   3065.242680  16180.075700   

      RANGEDISFERR   MEANDISFERR   STDDISFERR  
0     24858.149460  32835.065130  5375.348561  
1     16508.171420   5313.377212  3528.866119  
2     19156.101350  16201.363920  4873.245249  
3      3326.206333   1191.538806   687.041000  
4      4467.090147   4517.535001  1028.025195  
...            ...           ...          ...  
3926  10279.509740  16677.655730  2380.118939  
3927   4573.515029   4861.438266  1110.592797  
3928   4645.730343  15103.415760  1046.244374  
3929  12625.922140  15677.993590  2972.601362  
3930  13114.833020   9150.936673  2792.152885  

[3931 rows x 31 columns]

Ahora veamos qué municipios están contenidos en el csv que acabamos de crear.

In [26]:
excel110['CODMUNI'].sort_values()

2488     1001
2491     1002
2489     1003
2490     1004
2492     1006
        ...  
3214    53077
3009    53078
3016    53080
3017    53081
2747    53083
Name: CODMUNI, Length: 3931, dtype: int64

Repetimos la misma operación para el resto de csv.

**Excel 02:**

In [27]:
excel02 = excel02.rename(columns={"CODMUN":"CODMUNI"}) #Renombramos a CODMUNI para poder unir los archivos.

In [28]:
excel110['CODMUNI'].isin(excel02['CODMUNI']).value_counts()

True     3856
False      75
Name: CODMUNI, dtype: int64

Hay 3856 valores que están en ambos csv y 75 valores que sólo están en uno de los csv.

Unimos los csv:

In [29]:
excel110_02 = pd.merge(excel110_112, excel02, on='CODMUNI')

In [30]:
excel110_02

CODMUNI  MINMDT  MAXMDT  RANGEMDT      MEANMDT      STDMDT  MINPENDI  \
0       33001     117    1394      1277   784.735210  233.924492  0.173360   
1       33002     271    2105      1834  1067.056635  382.986501  0.122584   
2       33003      99    2393      2294   817.504449  455.771315  0.346717   
3       33004      -6     131       137    41.899237   34.009653  0.000000   
4       33006     274    1113       839   506.502204  142.357363  0.122584   
...       ...     ...     ...       ...          ...         ...       ...   
3851    50293     671    1408       737   896.568072  140.151654  0.000000   
3852    50294     890     979        89   937.295656   17.358179  0.000000   
3853    50902     342     443       101   385.815187   19.015430  0.000000   
3854    50901     616    1443       827   870.803159  148.132527  0.122584   
3855    50903     201     461       260   306.537509   68.091967  0.000000   

       MAXPENDI  RANGEPENDI  MEANPENDI  ...    TCVEG2011    TCVEG2012  \
0     46.501644   46.328284  20.298575  ...   -74.468085   -72.972973   
1     61.203892   61.081307  22.556711  ...  -377.551020  -478.417266   
2     67.186058   66.839341  25.832286  ...   -63.694268  -131.034483   
3     21.192459   21.192459   4.466160  ... -1040.983607 -1175.781250   
4     43.220619   43.098035  15.395220  ...  -232.323232  -319.148936   
...         ...         ...        ...  ...          ...          ...   
3851  32.033096   32.033096   6.590386  ...   -39.130435   -43.478261   
3852  12.097475   12.097475   2.596466  ...   -40.000000   -38.461538   
3853  12.974638   12.974638   2.771706  ...     0.000000     0.000000   
3854  39.169685   39.047101  11.458718  ...    -6.024096   -23.809524   
3855  11.338626   11.338626   2.532540  ...    -4.506066    -5.637773   

        TCVEG2013    TCVEG2014    TCVEG2015    TCVEG2016    TCVEG2017  \
0      -73.972603   -64.066852   -70.460705   -75.418994   -93.750000   
1     -539.325843  -452.107280  -674.698795  -793.388430  -554.166667   
2     -113.636364  -133.333333  -151.785714  -156.521739  -132.743363   
3    -1665.254237 -1310.077519 -1606.837607 -1798.283262 -2317.535545   
4     -188.888889  -178.947368  -234.042553  -265.957447  -271.739130   
...           ...          ...          ...          ...          ...   
3851   -50.228310   -33.195021   -68.825911   -37.190083   -38.461538   
3852     0.000000   -42.553191   -66.666667   -20.833333    40.000000   
3853     0.000000   -20.618557   -21.276596   -32.967033     0.000000   
3854    -5.988024    -6.369427   -13.513514   -20.979021   -37.593985   
3855    -1.409940    -3.546099    -5.728607    -5.448602    -2.205072   

        TCVEG2018    TCVEG2019    TCVEG2020  
0      -87.463557   -82.111437   -64.705882  
1     -630.705394  -579.166667  -678.111588  
2     -155.339806  -120.370370  -132.075472  
3    -2578.475336 -2210.084034 -2982.456140  
4     -180.851064  -153.846154  -228.571429  
...           ...          ...          ...  
3851   -51.094891   -29.197080   -25.641026  
3852     0.000000   -61.224490   -62.500000  
3853     0.000000   -34.883721   -23.809524  
3854     0.000000    -6.493506    -6.451613  
3855     0.370920     1.102941    -1.457195  

[3856 rows x 159 columns]

**Excel03:**

In [31]:
excel03 = excel03.rename(columns={"CODMUN":"CODMUNI"})

In [32]:
excel110['CODMUNI'].isin(excel03['CODMUNI']).value_counts()

True     3856
False      75
Name: CODMUNI, dtype: int64

Hay 3856 valores que están en ambos csv y 75 valores que sólo están en uno de los csv.

Unimos los csv:

In [33]:
excel110_03 = pd.merge(excel110_02, excel03, on='CODMUNI')

In [34]:
excel110_03

CODMUNI  MINMDT  MAXMDT  RANGEMDT      MEANMDT      STDMDT  MINPENDI  \
0       33001     117    1394      1277   784.735210  233.924492  0.173360   
1       33002     271    2105      1834  1067.056635  382.986501  0.122584   
2       33003      99    2393      2294   817.504449  455.771315  0.346717   
3       33004      -6     131       137    41.899237   34.009653  0.000000   
4       33006     274    1113       839   506.502204  142.357363  0.122584   
...       ...     ...     ...       ...          ...         ...       ...   
3851    50293     671    1408       737   896.568072  140.151654  0.000000   
3852    50294     890     979        89   937.295656   17.358179  0.000000   
3853    50902     342     443       101   385.815187   19.015430  0.000000   
3854    50901     616    1443       827   870.803159  148.132527  0.122584   
3855    50903     201     461       260   306.537509   68.091967  0.000000   

       MAXPENDI  RANGEPENDI  MEANPENDI  ...  SALDOEXT2019  ALTASTT2020  \
0     46.501644   46.328284  20.298575  ...             4           62   
1     61.203892   61.081307  22.556711  ...             4          331   
2     67.186058   66.839341  25.832286  ...             1           30   
3     21.192459   21.192459   4.466160  ...           394         2240   
4     43.220619   43.098035  15.395220  ...             3          110   
...         ...         ...        ...  ...           ...          ...   
3851  32.033096   32.033096   6.590386  ...             5           25   
3852  12.097475   12.097475   2.596466  ...             6           10   
3853  12.974638   12.974638   2.771706  ...             0           10   
3854  39.169685   39.047101  11.458718  ...             5           21   
3855  11.338626   11.338626   2.532540  ...            13          133   

      ALTASINT2020  ALTASEXT2020  BAJASTT2020  BAJASINT2020  BAJASEXT2020  \
0               56             6           39            39             0   
1              309            22          250           232            18   
2               23             7           20            20             0   
3             1897           343         2314          2159           155   
4              106             4           72            69             3   
...            ...           ...          ...           ...           ...   
3851            24             1            8             8             0   
3852             9             1            7             7             0   
3853            10             0            1             1             0   
3854            16             5            5             5             0   
3855           130             3          126           122             4   

      SALDOTT2020  SALDOINT2020  SALDOEXT2020  
0              23            17             6  
1              81            77             4  
2              10             3             7  
3             -74          -262           188  
4              38            37             1  
...           ...           ...           ...  
3851           17            16             1  
3852            3             2             1  
3853            9             9             0  
3854           16            11             5  
3855            7             8            -1  

[3856 rows x 241 columns]

**Excel04:**

In [35]:
excel04 = excel04.rename(columns={"CODMUN":"CODMUNI"})

In [36]:
excel110_04 = pd.merge(excel110_03, excel04, on='CODMUNI')

**Excel05:**

In [37]:
excel05 = excel05.rename(columns={"CODMUN":"CODMUNI"})

Miramos las columnas están repetidas en el csv y las eliminamos: 

In [38]:
for col_name in excel05: 
    print(col_name)

OBJECTID
CODMUNI
CCAA
NOMBRE
CODPROV
PROVINCIA
NombreCCAA
Shape_Length
Shape_Area
NombreProv
MUNICIPIO
POB2021
POB2020
POB2019
POB2018
POB2017
POB2016
POB2015
POB2014
POB2013
POB2012
POB2011
POB2010
POB2009
POB2008
POB2007
POB2006
POB2005
POB2004
POB2003
POB2002
POB2001
POB2000
POB1999
POB1998
POB1997
POB1996
ObjectID
POB1991
POB1981
POB1970
POB1960
POB1950
POB1940
POB1930
POB1920
POB1910
POB1900
AREAKM2
DEN1900
DEN1910
DEN1920
DEN1930
DEN1940
DEN1950
DEN1960
DEN1970
DEN1981
DEN1991
DEN1996
DEN1997
DEN1998
DEN1999
DEN2000
DEN2001
DEN2002
DEN2003
DEN2004
DEN2005
DEN2006
DEN2007
DEN2008
DEN2009
DEN2010
DEN2011
DEN2012
DEN2013
DEN2014
DEN2015
DEN2016
DEN2017
DEN2018
DEN2019
DEN2020
DEN2021


In [39]:
excel05 = excel05.drop(['OBJECTID'], axis=1) #Eliminamos las columnas repetidas.
excel05 = excel05.drop(['CCAA'], axis=1)
excel05 = excel05.drop(['NOMBRE'], axis=1)
excel05 = excel05.drop(['CODPROV'], axis=1)
excel05 = excel05.drop(['PROVINCIA'], axis=1)
excel05 = excel05.drop(['NombreCCAA'], axis=1)
excel05 = excel05.drop(['MUNICIPIO'], axis=1)
excel05 = excel05.drop(['ObjectID'], axis=1)
excel05 = excel05.drop(['NombreProv'], axis=1)

Una vez realizado este paso, sólo nos queda unirlo con el csv anterior para conseguir el excel con todas las variables.

In [40]:
excel_unificado1 = pd.merge(excel110_04, excel05, on='CODMUNI')

**Excel 12:**

In [41]:
excel12 = excel12.rename(columns={"codMuni":"CODMUNI"}) #Renombramos la columna CODMUN A CODMUNI para que ambas tengan el mismo nombre.

In [42]:
excel_unificado2 = pd.merge(excel_unificado1, excel12, on='CODMUNI')

**Excel 11:**

In [43]:
excel11 = excel11.rename(columns={"CODMUN":"CODMUNI"})

In [44]:
excel_unificado3 = pd.merge(excel_unificado2, excel11, on='CODMUNI')

**Excel 07:**

In [45]:
excel07 = excel07.rename(columns={"CODMUN":"CODMUNI"})

In [46]:
excel_unificado = pd.merge(excel_unificado3, excel07, on='CODMUNI')

In [47]:
excel_unificado = excel_unificado.dropna()
excel_unificado

CODMUNI  MINMDT  MAXMDT  RANGEMDT      MEANMDT      STDMDT  MINPENDI  \
0       33001     117    1394      1277   784.735210  233.924492  0.173360   
1       33002     271    2105      1834  1067.056635  382.986501  0.122584   
2       33003      99    2393      2294   817.504449  455.771315  0.346717   
3       33004      -6     131       137    41.899237   34.009653  0.000000   
4       33006     274    1113       839   506.502204  142.357363  0.122584   
...       ...     ...     ...       ...          ...         ...       ...   
3847    50292     831    1205       374   931.972503   73.191988  0.000000   
3848    50295     722    1129       407   863.795376   71.596410  0.173360   
3849    50297     176     736       560   334.380306  107.806258  0.000000   
3850    50298     241     770       529   413.819021  110.728010  0.000000   
3851    50293     671    1408       737   896.568072  140.151654  0.000000   

       MAXPENDI  RANGEPENDI  MEANPENDI  ...     DISTCURGH   DISTATPRIM  \
0     46.501644   46.328284  20.298575  ...   5997.693739  5274.955751   
1     61.203892   61.081307  22.556711  ...  11748.784830  3961.940782   
2     67.186058   66.839341  25.832286  ...   8966.275234  4285.074165   
3     21.192459   21.192459   4.466160  ...   1497.922036   798.311975   
4     43.220619   43.098035  15.395220  ...   5417.955543  2496.420733   
...         ...         ...        ...  ...           ...          ...   
3847  26.876348   26.876348   7.047567  ...  13996.062990  2585.646631   
3848  34.187431   34.014071  10.970295  ...   6210.190730  1943.478095   
3849  34.648865   34.648865   3.399286  ...   7293.752966  4435.893761   
3850  27.790625   27.790625   3.889147  ...   8860.678727  5572.568876   
3851  32.033096   32.033096   6.590386  ...   4471.936801  3520.124977   

      VIVTUR21  PLA21   KM2BLAN  KM2GRIS  CBMAS30  CBMAS100  COB3G  COB4G  
0            2     13  0.001683        0     0.00      0.00   0.96   0.94  
1           58    264  0.026321        0     1.00      0.00   1.00   1.00  
2           21    112  0.000740        0     0.30      0.00   0.95   0.69  
3           71    315  0.000344        0     1.00      1.00   1.00   1.00  
4            9     51  0.005534        0     0.17      0.00   1.00   1.00  
...        ...    ...       ...      ...      ...       ...    ...    ...  
3847         0      0  0.000000        0     0.95      0.00   1.00   1.00  
3848         0      0  0.002940        0     0.95      0.00   0.98   1.00  
3849       659   2614  0.000000        0     0.99      0.99   1.00   1.00  
3850         1      2  0.415539        0     0.95      0.02   1.00   1.00  
3851         0      0  0.000139        0     0.86      0.86   1.00   1.00  

[3714 rows x 587 columns]

Veamos, nuevamente, los valores de las columnas, para ver si hay algún valor repetido. Para que la búsqueda sea más fácil, se pueden ordenar las variables mediante el comando excel_unificado.reindex(sorted(excel_unificado.columns).

In [48]:
for col_name in excel_unificado.reindex(sorted(excel_unificado.columns), axis=1): 
    print(col_name)

0A4_03
0A4_10
0A4_21
100YMS_03
100YMS_10
100YMS_21
10A14_03
10A14_10
10A14_21
15A19_03
15A19_10
15A19_21
20A24_03
20A24_10
20A24_21
25A29_03
25A29_10
25A29_21
30A34_03
30A34_10
30A34_21
35A39_03
35A39_10
35A39_21
40A44_03
40A44_10
40A44_21
45A49_03
45A49_10
45A49_21
50A54_03
50A54_10
50A54_21
55A59_03
55A59_10
55A59_21
5A9_03
5A9_10
5A9_21
60A64_03
60A64_10
60A64_21
65A69_03
65A69_10
65A69_21
70A74_03
70A74_10
70A74_21
75A79_03
75A79_10
75A79_21
80A84_03
80A84_10
80A84_21
85A89_03
85A89_10
85A89_21
90A94_03
90A94_10
90A94_21
95A99_03
95A99_10
95A99_21
ALTASEXT2012
ALTASEXT2013
ALTASEXT2014
ALTASEXT2015
ALTASEXT2016
ALTASEXT2017
ALTASEXT2018
ALTASEXT2019
ALTASEXT2020
ALTASINT2012
ALTASINT2013
ALTASINT2014
ALTASINT2015
ALTASINT2016
ALTASINT2017
ALTASINT2018
ALTASINT2019
ALTASINT2020
ALTASTT2012
ALTASTT2013
ALTASTT2014
ALTASTT2015
ALTASTT2016
ALTASTT2017
ALTASTT2018
ALTASTT2019
ALTASTT2020
AREAKM2
BAJASEXT2012
BAJASEXT2013
BAJASEXT2014
BAJASEXT2015
BAJASEXT2016
BAJASEXT2017
BAJASEXT2018
B

Eliminamos la variable "NOMBRE", por ser la misma variable que "Provincia".

In [49]:
excel_unificado = excel_unificado.drop(['NOMBRE'], axis=1)

Comprobamos que se han eliminado los valores y que no queda ninguna columna repetida.

In [50]:
for col_name in excel_unificado.reindex(sorted(excel_unificado.columns), axis=1): 
    print(col_name)

0A4_03
0A4_10
0A4_21
100YMS_03
100YMS_10
100YMS_21
10A14_03
10A14_10
10A14_21
15A19_03
15A19_10
15A19_21
20A24_03
20A24_10
20A24_21
25A29_03
25A29_10
25A29_21
30A34_03
30A34_10
30A34_21
35A39_03
35A39_10
35A39_21
40A44_03
40A44_10
40A44_21
45A49_03
45A49_10
45A49_21
50A54_03
50A54_10
50A54_21
55A59_03
55A59_10
55A59_21
5A9_03
5A9_10
5A9_21
60A64_03
60A64_10
60A64_21
65A69_03
65A69_10
65A69_21
70A74_03
70A74_10
70A74_21
75A79_03
75A79_10
75A79_21
80A84_03
80A84_10
80A84_21
85A89_03
85A89_10
85A89_21
90A94_03
90A94_10
90A94_21
95A99_03
95A99_10
95A99_21
ALTASEXT2012
ALTASEXT2013
ALTASEXT2014
ALTASEXT2015
ALTASEXT2016
ALTASEXT2017
ALTASEXT2018
ALTASEXT2019
ALTASEXT2020
ALTASINT2012
ALTASINT2013
ALTASINT2014
ALTASINT2015
ALTASINT2016
ALTASINT2017
ALTASINT2018
ALTASINT2019
ALTASINT2020
ALTASTT2012
ALTASTT2013
ALTASTT2014
ALTASTT2015
ALTASTT2016
ALTASTT2017
ALTASTT2018
ALTASTT2019
ALTASTT2020
AREAKM2
BAJASEXT2012
BAJASEXT2013
BAJASEXT2014
BAJASEXT2015
BAJASEXT2016
BAJASEXT2017
BAJASEXT2018
B

Resultado de la unión de todos los csv:

In [51]:
excel_unificado

CODMUNI  MINMDT  MAXMDT  RANGEMDT      MEANMDT      STDMDT  MINPENDI  \
0       33001     117    1394      1277   784.735210  233.924492  0.173360   
1       33002     271    2105      1834  1067.056635  382.986501  0.122584   
2       33003      99    2393      2294   817.504449  455.771315  0.346717   
3       33004      -6     131       137    41.899237   34.009653  0.000000   
4       33006     274    1113       839   506.502204  142.357363  0.122584   
...       ...     ...     ...       ...          ...         ...       ...   
3847    50292     831    1205       374   931.972503   73.191988  0.000000   
3848    50295     722    1129       407   863.795376   71.596410  0.173360   
3849    50297     176     736       560   334.380306  107.806258  0.000000   
3850    50298     241     770       529   413.819021  110.728010  0.000000   
3851    50293     671    1408       737   896.568072  140.151654  0.000000   

       MAXPENDI  RANGEPENDI  MEANPENDI  ...     DISTCURGH   DISTATPRIM  \
0     46.501644   46.328284  20.298575  ...   5997.693739  5274.955751   
1     61.203892   61.081307  22.556711  ...  11748.784830  3961.940782   
2     67.186058   66.839341  25.832286  ...   8966.275234  4285.074165   
3     21.192459   21.192459   4.466160  ...   1497.922036   798.311975   
4     43.220619   43.098035  15.395220  ...   5417.955543  2496.420733   
...         ...         ...        ...  ...           ...          ...   
3847  26.876348   26.876348   7.047567  ...  13996.062990  2585.646631   
3848  34.187431   34.014071  10.970295  ...   6210.190730  1943.478095   
3849  34.648865   34.648865   3.399286  ...   7293.752966  4435.893761   
3850  27.790625   27.790625   3.889147  ...   8860.678727  5572.568876   
3851  32.033096   32.033096   6.590386  ...   4471.936801  3520.124977   

      VIVTUR21  PLA21   KM2BLAN  KM2GRIS  CBMAS30  CBMAS100  COB3G  COB4G  
0            2     13  0.001683        0     0.00      0.00   0.96   0.94  
1           58    264  0.026321        0     1.00      0.00   1.00   1.00  
2           21    112  0.000740        0     0.30      0.00   0.95   0.69  
3           71    315  0.000344        0     1.00      1.00   1.00   1.00  
4            9     51  0.005534        0     0.17      0.00   1.00   1.00  
...        ...    ...       ...      ...      ...       ...    ...    ...  
3847         0      0  0.000000        0     0.95      0.00   1.00   1.00  
3848         0      0  0.002940        0     0.95      0.00   0.98   1.00  
3849       659   2614  0.000000        0     0.99      0.99   1.00   1.00  
3850         1      2  0.415539        0     0.95      0.02   1.00   1.00  
3851         0      0  0.000139        0     0.86      0.86   1.00   1.00  

[3714 rows x 586 columns]

Posicionamos las variables más relevantes para la identificación de los municipios al principio del dataset. Estas son las variables de:

1. **ObjectID:** *Código asignado para establecer relaciones entre los datos y otras entidades en un entorno de Sistemas de    Información Geográfica.*
2. **NombreCCAA:** *Nombre de la Comunidad Autónoma.*
3. **PROVINCIA:** *Nombre de la provincia junto a su código identificador del INE.*
4. **MUNICIPIO:** *Nombre del municipio junto a su código identificador del INE.*

In [52]:
excel_unificado = excel_unificado[ ['CODMUNI'] + ['OBJECTID'] + ['NombreCCAA'] + ['Provincia'] + ['Municipio'] + [ col for col in excel_unificado.columns if col !='CODMUNI' and col != 'OBJECTID' and col != 'NombreCCAA' and col != 'Provincia' and col != 'Municipio']]
excel_unificado

CODMUNI  OBJECTID              NombreCCAA Provincia  \
0       33001         1  Principado de Asturias  Asturias   
1       33002         2  Principado de Asturias  Asturias   
2       33003         3  Principado de Asturias  Asturias   
3       33004         4  Principado de Asturias  Asturias   
4       33006         5  Principado de Asturias  Asturias   
...       ...       ...                     ...       ...   
3847    50292      3923                  Aragón  Zaragoza   
3848    50295      3924                  Aragón  Zaragoza   
3849    50297      3925                  Aragón  Zaragoza   
3850    50298      3926                  Aragón  Zaragoza   
3851    50293      3927                  Aragón  Zaragoza   

                    Municipio  MINMDT  MAXMDT  RANGEMDT      MEANMDT  \
0                     Allande     117    1394      1277   784.735210   
1                       Aller     271    2105      1834  1067.056635   
2                      Amieva      99    2393      2294   817.504449   
3                      Avilés      -6     131       137    41.899237   
4                     Bimenes     274    1113       839   506.502204   
...                       ...     ...     ...       ...          ...   
3847     Villarreal de Huerva     831    1205       374   931.972503   
3848               Vistabella     722    1129       407   863.795376   
3849                 Zaragoza     176     736       560   334.380306   
3850                    Zuera     241     770       529   413.819021   
3851  Villarroya de la Sierra     671    1408       737   896.568072   

          STDMDT  ...     DISTCURGH   DISTATPRIM  VIVTUR21  PLA21   KM2BLAN  \
0     233.924492  ...   5997.693739  5274.955751         2     13  0.001683   
1     382.986501  ...  11748.784830  3961.940782        58    264  0.026321   
2     455.771315  ...   8966.275234  4285.074165        21    112  0.000740   
3      34.009653  ...   1497.922036   798.311975        71    315  0.000344   
4     142.357363  ...   5417.955543  2496.420733         9     51  0.005534   
...          ...  ...           ...          ...       ...    ...       ...   
3847   73.191988  ...  13996.062990  2585.646631         0      0  0.000000   
3848   71.596410  ...   6210.190730  1943.478095         0      0  0.002940   
3849  107.806258  ...   7293.752966  4435.893761       659   2614  0.000000   
3850  110.728010  ...   8860.678727  5572.568876         1      2  0.415539   
3851  140.151654  ...   4471.936801  3520.124977         0      0  0.000139   

      KM2GRIS  CBMAS30  CBMAS100  COB3G  COB4G  
0           0     0.00      0.00   0.96   0.94  
1           0     1.00      0.00   1.00   1.00  
2           0     0.30      0.00   0.95   0.69  
3           0     1.00      1.00   1.00   1.00  
4           0     0.17      0.00   1.00   1.00  
...       ...      ...       ...    ...    ...  
3847        0     0.95      0.00   1.00   1.00  
3848        0     0.95      0.00   0.98   1.00  
3849        0     0.99      0.99   1.00   1.00  
3850        0     0.95      0.02   1.00   1.00  
3851        0     0.86      0.86   1.00   1.00  

[3714 rows x 586 columns]

### Filtrado de datos. Ciudades con población menor a 50000 habitantes:

Seleccionamos los municipios con menos de 50000 habitantes, tanto para el año 2020 como para el 2021. Un número elevado de habitantes provocaría que los resultados de la red neuronal no fuesen los óptimos.

In [53]:
excel_pob = excel_unificado[excel_unificado['POB2021'] < 50000]
excel_pob

CODMUNI  OBJECTID              NombreCCAA Provincia  \
0       33001         1  Principado de Asturias  Asturias   
1       33002         2  Principado de Asturias  Asturias   
2       33003         3  Principado de Asturias  Asturias   
4       33006         5  Principado de Asturias  Asturias   
5       33007         6  Principado de Asturias  Asturias   
...       ...       ...                     ...       ...   
3846    50296      3922                  Aragón  Zaragoza   
3847    50292      3923                  Aragón  Zaragoza   
3848    50295      3924                  Aragón  Zaragoza   
3850    50298      3926                  Aragón  Zaragoza   
3851    50293      3927                  Aragón  Zaragoza   

                    Municipio  MINMDT  MAXMDT  RANGEMDT      MEANMDT  \
0                     Allande     117    1394      1277   784.735210   
1                       Aller     271    2105      1834  1067.056635   
2                      Amieva      99    2393      2294   817.504449   
4                     Bimenes     274    1113       839   506.502204   
5                        Boal      33    1151      1118   493.553497   
...                       ...     ...     ...       ...          ...   
3846               Zaida (La)     139     300       161   189.077045   
3847     Villarreal de Huerva     831    1205       374   931.972503   
3848               Vistabella     722    1129       407   863.795376   
3850                    Zuera     241     770       529   413.819021   
3851  Villarroya de la Sierra     671    1408       737   896.568072   

          STDMDT  ...     DISTCURGH   DISTATPRIM  VIVTUR21  PLA21   KM2BLAN  \
0     233.924492  ...   5997.693739  5274.955751         2     13  0.001683   
1     382.986501  ...  11748.784830  3961.940782        58    264  0.026321   
2     455.771315  ...   8966.275234  4285.074165        21    112  0.000740   
4     142.357363  ...   5417.955543  2496.420733         9     51  0.005534   
5     218.406880  ...   4317.579792  4305.153512        15     72  0.018779   
...          ...  ...           ...          ...       ...    ...       ...   
3846   36.271257  ...   7899.678143  2227.740226         0      0  0.000000   
3847   73.191988  ...  13996.062990  2585.646631         0      0  0.000000   
3848   71.596410  ...   6210.190730  1943.478095         0      0  0.002940   
3850  110.728010  ...   8860.678727  5572.568876         1      2  0.415539   
3851  140.151654  ...   4471.936801  3520.124977         0      0  0.000139   

      KM2GRIS  CBMAS30  CBMAS100  COB3G  COB4G  
0           0     0.00      0.00   0.96   0.94  
1           0     1.00      0.00   1.00   1.00  
2           0     0.30      0.00   0.95   0.69  
4           0     0.17      0.00   1.00   1.00  
5           0     0.75      0.00   1.00   1.00  
...       ...      ...       ...    ...    ...  
3846        0     1.00      0.86   1.00   1.00  
3847        0     0.95      0.00   1.00   1.00  
3848        0     0.95      0.00   0.98   1.00  
3850        0     0.95      0.02   1.00   1.00  
3851        0     0.86      0.86   1.00   1.00  

[3689 rows x 586 columns]

In [54]:
excel_pob = excel_unificado[excel_unificado['POB2020'] < 50000]
excel_pob

CODMUNI  OBJECTID              NombreCCAA Provincia  \
0       33001         1  Principado de Asturias  Asturias   
1       33002         2  Principado de Asturias  Asturias   
2       33003         3  Principado de Asturias  Asturias   
4       33006         5  Principado de Asturias  Asturias   
5       33007         6  Principado de Asturias  Asturias   
...       ...       ...                     ...       ...   
3846    50296      3922                  Aragón  Zaragoza   
3847    50292      3923                  Aragón  Zaragoza   
3848    50295      3924                  Aragón  Zaragoza   
3850    50298      3926                  Aragón  Zaragoza   
3851    50293      3927                  Aragón  Zaragoza   

                    Municipio  MINMDT  MAXMDT  RANGEMDT      MEANMDT  \
0                     Allande     117    1394      1277   784.735210   
1                       Aller     271    2105      1834  1067.056635   
2                      Amieva      99    2393      2294   817.504449   
4                     Bimenes     274    1113       839   506.502204   
5                        Boal      33    1151      1118   493.553497   
...                       ...     ...     ...       ...          ...   
3846               Zaida (La)     139     300       161   189.077045   
3847     Villarreal de Huerva     831    1205       374   931.972503   
3848               Vistabella     722    1129       407   863.795376   
3850                    Zuera     241     770       529   413.819021   
3851  Villarroya de la Sierra     671    1408       737   896.568072   

          STDMDT  ...     DISTCURGH   DISTATPRIM  VIVTUR21  PLA21   KM2BLAN  \
0     233.924492  ...   5997.693739  5274.955751         2     13  0.001683   
1     382.986501  ...  11748.784830  3961.940782        58    264  0.026321   
2     455.771315  ...   8966.275234  4285.074165        21    112  0.000740   
4     142.357363  ...   5417.955543  2496.420733         9     51  0.005534   
5     218.406880  ...   4317.579792  4305.153512        15     72  0.018779   
...          ...  ...           ...          ...       ...    ...       ...   
3846   36.271257  ...   7899.678143  2227.740226         0      0  0.000000   
3847   73.191988  ...  13996.062990  2585.646631         0      0  0.000000   
3848   71.596410  ...   6210.190730  1943.478095         0      0  0.002940   
3850  110.728010  ...   8860.678727  5572.568876         1      2  0.415539   
3851  140.151654  ...   4471.936801  3520.124977         0      0  0.000139   

      KM2GRIS  CBMAS30  CBMAS100  COB3G  COB4G  
0           0     0.00      0.00   0.96   0.94  
1           0     1.00      0.00   1.00   1.00  
2           0     0.30      0.00   0.95   0.69  
4           0     0.17      0.00   1.00   1.00  
5           0     0.75      0.00   1.00   1.00  
...       ...      ...       ...    ...    ...  
3846        0     1.00      0.86   1.00   1.00  
3847        0     0.95      0.00   1.00   1.00  
3848        0     0.95      0.00   0.98   1.00  
3850        0     0.95      0.02   1.00   1.00  
3851        0     0.86      0.86   1.00   1.00  

[3689 rows x 586 columns]

Hay un total de 3689 municipios con una población inferior a 50000 habitantes.

### Filtrado de datos para Cantabria:

Seleccionamos los datos para la Comunidad de Cantabria.

In [55]:
excel_cantabria = excel_pob[excel_pob['NombreCCAA'] == 'Cantabria'] 
excel_cantabria

CODMUNI  OBJECTID NombreCCAA  Provincia              Municipio  MINMDT  \
2326    39084      2377  Cantabria  Cantabria              Solórzano      65   
2327    39085      2378  Cantabria  Cantabria                Suances      -6   
2328    39001      2379  Cantabria  Cantabria       Alfoz de Lloredo      -3   
2329    39003      2380  Cantabria  Cantabria                Anievas     203   
2330    39100      2381  Cantabria  Cantabria             Villafufre     112   
...       ...       ...        ...        ...                    ...     ...   
2423    39097      2474  Cantabria  Cantabria            Vega de Pas     247   
2424    39098      2475  Cantabria  Cantabria          Villacarriedo     150   
2425    39099      2476  Cantabria  Cantabria            Villaescusa       0   
2426    39101      2477  Cantabria  Cantabria  Villaverde de Trucíos     152   
2427    39102      2478  Cantabria  Cantabria                   Voto      -3   

      MAXMDT  RANGEMDT     MEANMDT      STDMDT  ...    DISTCURGH   DISTATPRIM  \
2326     552       487  220.690553  104.095285  ...  8655.201148  1503.589655   
2327     218       224   57.649513   42.324675  ...  2249.691385  1682.295735   
2328     391       394  141.156113   83.474135  ...  7292.339080  1657.810817   
2329     942       739  467.784320  140.755881  ...  6239.139244  1794.658987   
2330     828       716  329.966152  142.751930  ...  6234.089421  2070.778547   
...      ...       ...         ...         ...  ...          ...          ...   
2423    1682      1435  716.320500  251.176248  ...  9058.263289  4048.590984   
2424     870       720  429.996782  182.276557  ...  3363.345133  2203.640557   
2425     437       437  112.661945   80.694858  ...  3405.989670  1269.354330   
2426     735       583  363.969076  136.276553  ...  6938.838735  2412.441843   
2427     808       811  217.173197  167.650140  ...  6154.359040  2932.312878   

      VIVTUR21  PLA21   KM2BLAN  KM2GRIS  CBMAS30  CBMAS100  COB3G  COB4G  
2326         9     79  0.020988        0     0.60      0.00   0.98   0.98  
2327       156    814  0.000103        0     0.99      0.62   1.00   1.00  
2328        64    322  0.004960        0     0.96      0.00   1.00   1.00  
2329         5     33  0.025491        0     0.00      0.00   1.00   1.00  
2330         3     10  0.024887        0     0.98      0.02   1.00   1.00  
...        ...    ...       ...      ...      ...       ...    ...    ...  
2423        10     50  0.183440        0     0.29      0.00   1.00   1.00  
2424        23    169  0.058897        0     1.00      1.00   1.00   1.00  
2425        10     64  0.004778        0     0.98      0.79   1.00   1.00  
2426         0      0  0.000556        0     0.49      0.00   1.00   1.00  
2427        35    197  0.912579        0     1.00      0.00   1.00   1.00  

[100 rows x 586 columns]

Hay un total de 100 municipios y 586 variables.

## Guardado de Datasets

Finalmente, se guardan los datasets para que puedan ser utilizados en los siguientes análisis.

In [56]:
excel_pob.to_csv('excel_pob.csv', index=False)

In [57]:
excel_cantabria.to_csv('excel_cantabria.csv', index=False)